components/dashboard/ExportTransactionsCSVModal.tsx

```diff
@@ -42,32 +48,43 @@ import { Filterbar } from './filters/Filterbar';
 
 const env = process.env.OC_ENV;
 
+const getAccountSettingsQuery = gql`
+  query GetAccountSettings($slug: String!) {
+    account(slug: $slug) {
+      id
+      settings
+    }
+  }
+`;
+
 type ExportTransactionsCSVModalProps = {
   open?: boolean;
   setOpen?: (open: boolean) => void;
   queryFilter: useQueryFilterReturnType<any, TransactionsPageQueryVariables | HostReportsPageQueryVariables>;
-  accountSlug?: string;
-  hostSlug?: string;
+  account?: Pick<Account, 'slug'>;
+  isHostReport?: boolean;
   trigger: React.ReactNode;
 };
 
 const ExportTransactionsCSVModal = ({
   open,
   setOpen,
-  accountSlug,
-  hostSlug,
+  account,
   trigger,
   queryFilter,
+  isHostReport,
 }: ExportTransactionsCSVModalProps) => {
-  const isHostReport = Boolean(hostSlug);
-
   const [downloadUrl, setDownloadUrl] = React.useState<string | null>('#');
-  const [fieldOption, setFieldOption] = React.useState(FieldOptions[0].value);
+  const [fieldSet, setFieldSet] = React.useState(FieldOptions[0].value);
   const [fields, setFields] = React.useState(DEFAULT_FIELDS.reduce((obj, key) => ({ ...obj, [key]: true }), {}));
   const [flattenTaxesAndPaymentProcessorFees, setFlattenTaxesAndPaymentProcessorFees] = React.useState(false);
-
   const fieldGroups = FIELD_GROUPS_2024;
 
+  const { data: accountSettings } = useQuery(getAccountSettingsQuery, {
+    context: API_V2_CONTEXT,
+    variables: { slug: account?.slug },
+  });
+
```

Comment:
Wouldn't that be easier to make sur that settings is set on the `account` that is passed pas props instead of reloading from GraphQL ?

Reply:
Sure, I think that's the way to go here... I'll update the type of this component to require settings and update the other requests.